lib/src/Uncertainty/Algorithm/MetaModel/MetaModelAlgorithm.cxx

```diff
@@ -60,6 +68,141 @@ String MetaModelAlgorithm::__repr__() const
   return oss;
 }
 
+struct TestedDistribution
+{
+  Distribution distribution_;
+  Scalar score_;
+  Scalar bic_;
+  Scalar aic_;
+  Scalar aicc_;
+  String criterion_;
+
+  TestedDistribution(const Distribution & distribution,
+                     const Scalar score,
+                     const Scalar bic,
+                     const Scalar aic,
+                     const Scalar aicc,
+                     const String & criterion)
+  : distribution_(distribution)
+  , score_(score)
+  , bic_(bic)
+  , aic_(aic)
+  , aicc_(aicc)
+  , criterion_(criterion)
+  {
+    // Nothing to do
+  }
+
+  bool operator< (const TestedDistribution& other) const
+  {
+    if (criterion_ == "BIC")
+      return bic_ < other.bic_;
+    else if (criterion_ == "AIC")
+      return aic_ < other.aic_;
+    else if (criterion_ == "AICC")
+      return aicc_ < other.aicc_;
+    else
+      return score_ > other.score_;
+  }
+};
+
+Distribution MetaModelAlgorithm::BuildDistribution(const Sample & inputSample)
+{
+  // Recover the distribution, taking into account that we look for performance
+  // so we avoid to rebuild expensive distributions as much as possible
+  const UnsignedInteger inputDimension = inputSample.getDimension();
+  Collection< Distribution > marginals(inputDimension);
+  // The strategy for the marginals is to find the best continuous 1-d parametric model else fallback to a non-parametric one
+  DistributionFactory nonParametricModel;
+  const String nonParametric = ResourceMap::GetAsString("MetaModelAlgorithm-NonParametricModel");
+  if (nonParametric == "Histogram")
+    nonParametricModel = DistributionFactory(HistogramFactory());
+  else
+    nonParametricModel = DistributionFactory(KernelSmoothing());
+
+  Collection< DistributionFactory > factories(DistributionFactory::GetContinuousUniVariateFactories());
+  // Filter factories : remove histogram
+  for (UnsignedInteger i = 0; i < factories.getSize(); ++i)
+  {
+    if (factories[i].getImplementation()->getClassName() == "HistogramFactory")
+    {
+      factories.erase(factories.begin() + i);
+      break;
+    }
+  }
+
+  // Level for model qualification using KS test
+  const Scalar level = ResourceMap::GetAsScalar("MetaModelAlgorithm-PValueThreshold");
+  // Model selection using BIC/AIC/AICC criterion
+  const String criterion = ResourceMap::GetAsString("MetaModelAlgorithm-ModelSelectionCriterion");
+
+  const Description inputDescription(inputSample.getDescription());
+  for (UnsignedInteger i = 0; i < inputDimension; ++i)
+  {
+    // Here we remove the duplicate entries in the marginal sample as we are suppose to have a continuous distribution. 
+    // The duplicates are mostly due to truncation in the file export.
+    const Sample marginalSample(inputSample.getMarginal(i).sortUnique());
+    // First we estimate distribution using its factory
+    // Once it is possible, we keep it only if Kolmogorov score exceed level
+    std::vector<TestedDistribution> possibleDistributions;
```

Comment:
Why do you use a std::vector instead of a Collection here?

Reply:
I do not remember exactly the issue I had. So I moved to `std::vector`